src/dropdown-menu/dropdown-menu.component.jsx

```diff
@@ -1,82 +1,80 @@
-/* eslint-disable react/no-array-index-key */
-
 import React, { PropTypes } from 'react';
-import { DropdownButton, MenuItem } from 'react-bootstrap';
+import { MenuItem } from 'react-bootstrap';
+
+import { DropdownContainer } from '../dropdown-container/index';
 import { Icon } from '../icons';
 import './dropdown-menu.component.scss';
 
-export default class DropdownMenu extends React.Component {
+export default class DropdownMenu extends React.PureComponent {
 
   static propTypes = {
-    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
-    title: PropTypes.oneOfType([PropTypes.string, PropTypes.element, PropTypes.number]),
-    caret: PropTypes.bool,
+    id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
     menuItems: PropTypes.arrayOf(PropTypes.shape({
-      title: PropTypes.oneOfType([
-        PropTypes.string,
-        PropTypes.element,
-        PropTypes.number,
-      ]),
-      type: PropTypes.oneOf(['item', 'divider']),
-      icon: PropTypes.element,
       disabled: PropTypes.bool,
       disableClosing: PropTypes.bool,
       href: PropTypes.string,
+      icon: PropTypes.element,
+      id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]), // serves as a key
       onClick: PropTypes.func,
+      title: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.element]),
+      type: PropTypes.oneOf(['item', 'divider']),
     })).isRequired,
+    caret: PropTypes.bool,
     disabled: PropTypes.bool,
     dropup: PropTypes.bool,
     pullLeft: PropTypes.bool,
+    title: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.element]),
   };
 
   static defaultProps = {
-    title: <Icon type="indicator" name="more" width={32} height={32} />,
     caret: false,
     disabled: false,
     dropup: false,
     pullLeft: false,
+    title: <Icon type="indicator" name="more" width={32} height={32} />,
   };
 
   constructor(props) {
     super(props);
-    this.state = {
-      menuOpen: false,
-    };
+    this.state = { isOpen: false };
+  }
+
+  handleClick = (e, item) => {
+    if (item.disableClosing) {
+      this.dontCloseDropdownMenu = true;
+    }
+    if (!item.disabled && item.onClick !== undefined) {
+      item.onClick(e);
```

Comment:
This component doesn't render multi-select and multi-select-item component. I removed handleClick function and returned the previous code though.